### PR TITLE
Multi-language support & lazy command line & clean up

### PR DESCRIPTION
**Pull request by JPDF**
### Multi-language support
Added multi-language support which makes it easy to change language on the fly and add whatever language you want. To add a new language simply configure the languages.json file and add a corresponding template file.

### Lazy command line
Added a storage system to save command argument so you can type what language you want once and which problem you have once.
Example:
```
npm run init problem1 c
npm run test
npm run submit

npm run init problem2
npm run test
npm run submit
```
These can be overwritten as well if you want to test or submit some other problem file or w.e.
Example:
```
npm run init problem1 node
npm run test problem2 c
```
But the problem and language will only be saved in the `init` command

### Testing
Remade how testing shows on screen. Removed the assertion and made a custom output which in my opinion looks neat.
code_2019-02-02_23-21-13

### Logging changes
Added some error handling instead of relying on crashes, it now describes what's wrong to the user instead of showing stack traces.

### Clean up
Cleaned up some files to remove repeated code etc.

### Others
The settings.js file now handles the input, storage loading and loading the language.json file.
Added 'Run time error' to test cases with a new icon since that was apparently a thing...

### CODE REVIEW

File: languages.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (dinosaursarecool) [inline languages.json]: If we're defining languages here we should preferably reuse this definition for the submit script

**Comment by dinosaursarecool** [comment]
Fixes #1

**Review by dinosaursarecool — APPROVED** [review]
Nice one 👍

File: utils/constants.js [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
-	WRONG_ANSWER: '❌',
-	'not checked': '❔',
-	NOT_CHECKED: '❔',
+    'Accepted': '✅',
+    ACCEPTED: '✅',
+    'Wrong Answer': '❌',
+    WRONG_ANSWER: '❌',
+    'not checked': '❔',
+    NOT_CHECKED: '❔',
+    RUN_TIME_ERROR: '⚠️',
```
Comment (dinosaursarecool) [inline utils/constants.js]: Haha, speaking from experience I take it? 😉 😉 😉

File: utils/testHelper.js [hunk 3]
```diff
@@ -2,53 +2,92 @@ const path = require('path')
...
+        result['expectedOutput'] = normalizeNewline(expectedOutput)
+
+        if (Object.is(result['output'], result['expectedOutput']))
+            result['status'] = TEST_STATUS.ACCEPTED
+        else
+            result['status'] = TEST_STATUS.WRONG_ANSWER
+        count++;
+    }
+    let prettyOutput = '\x1B[31m- actual \x1B[32m+ expected\x1B[0m\n'
+    results.forEach(r => {
```
Comment (dinosaursarecool) [inline utils/testHelper.js]: could just `join` these instead of `forEach`ing to a non-const

**Comment by dinosaursarecool** [comment]
Thanks for the nice contribution! 🏅  Left some comments that I think would be nice to address, but you have my tentative ✔️ so go ahead and merge if you like. 

PS. Might I suggest a bit cleaner commit msgs that show the intent of the commits? That would make it more straight forward to review, because as of now a majority of the changes are in the single commit of doom (ece0391c9c633ed38b364083e0c3d23a46abd8a8), whereas the readme was granted several commits, although with the same msg (maybe either `git  amend` or add better descriptions imho)

File: utils/constants.js (hunk 2 above)
Comment (JPDF) [reply utils/constants.js]: Autoformat...😥

File: languages.json (hunk 1 above)
Comment (JPDF) [reply languages.json]: Can you elaborate?

**Comment by JPDF** [comment]
Thanks, I an not used with git, i wanted to have better commit msg but it just wanted a one liner so.. but i will definetly look into it.

PS. Sorry about the auto formatting, maybe you could share yours so i can keep it consistent

File: utils/testHelper.js (hunk 3 above)
Comment (JPDF) [reply utils/testHelper.js]: Will so!

**Comment by JPDF** [comment]
Also forgot to mention that if a file has been accepted on a submit it will move to the new accepted folder

File: utils/constants.js (hunk 2 above)
Comment (JPDF) [reply utils/constants.js]: Ah, I get what you mean now. Yes apparently run time error was a thing when they removed a problem which I was testing on...

**Pull request merged**